### PR TITLE
gocyclo: Update gocyclo to upstream v0.3.1

### PR DESCRIPTION
**Pull request by stackshadow**
###### Motivation for this change

Update gocyclo to a more recent version that support ignoring of vendor-directory and recursive iteration

###### Things done

The original repository used in this package was a fork from https://github.com/fzipp/gocyclo but the upstream is a way more updated. So i switch this package to https://github.com/fzipp/gocyclo

- Built on platform(s)
  - [x] x86_64-linux
  - aarch64-linux
  - x86_64-darwin
  - aarch64-darwin
- For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - (Module updates) Added a release notes entry if the change is significant
  - (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stackshadow** [comment]
What to do when nobody is there for an review ? 🤔

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 1]
```diff
@@ -392,6 +392,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The `services.ddclient.password` option was removed, and replaced with `services.ddclient.passwordFile`.
 
+- The `gocyclo` package is now based on the upstream version `v0.3.1` from https://github.com/fzipp/gocyclo
```
Comment (aanderse) [inline nixos/doc/manual/release-notes/rl-2111.section.md:395]: Please make links clickable and users will love you.

File: pkgs/development/tools/gocyclo/default.nix [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-    sha256 = "094rj97q38j53lmn2scshrg8kws8c542yq5apih1ahm9wdkv8pxr";
+    sha256 = "1rimdrhmy6nkzh7ydpx6139hh9ml6rdqg5gvkpy2l1x9mhanvan0";
   };
 
   meta = with lib; {
-    description = "Calculate cyclomatic complexities of functions in Go source code";
-    homepage = "https://github.com/alecthomas/gocyclo";
+    description =
+      "Calculate cyclomatic complexities of functions in Go source code";
+    homepage = "https://github.com/fzipp/gocyclo";
```
Comment (aanderse) [inline pkgs/development/tools/gocyclo/default.nix:21]: Please link to some statement that legitimized the change of source url here for future reference. I like to document that we're not slipping malicious software into nixpkgs when possible.

**Review by aanderse — CHANGES_REQUESTED** [review]
Just a few easy things to address please.

File: pkgs/development/tools/gocyclo/default.nix (hunk 2 above)
Comment (vs49688) [reply pkgs/development/tools/gocyclo/default.nix:21]: There's not really a statement for this, if you look at https://github.com/alecthomas/gocyclo, you'll see it's forked from https://github.com/fzipp/gocyclo, which is _vastly_ more up-to-date.

Perhaps just a note in the commit message?

Comment (aanderse) [reply pkgs/development/tools/gocyclo/default.nix:21]: So the project hasn't been transferred... just that a random developer forked it. Worth briefly mentioning that we're pulling sources from another authot now.

**Comment by stackshadow** [comment]
I will close this MR in favour for #157227

**Comment by stackshadow** [comment]
I will close this MR in favour for #157227

**Pull request closed without merging**